esmf_regrid/experimental/unstructured_regrid.py

```diff
@@ -116,13 +118,13 @@ def make_esmf_field(self):
 
     def size(self):
         """Return the number of cells in the mesh."""
-        return self.fnc.shape[0]
+        return self.shape[0]
 
     def _index_offset(self):
         return self.esi
 
     def _flatten_array(self, array):
-        return array
+        return array.reshape(-1, (self.size()), order="F").T
 
-    def _unflatten_array(self, array):
-        return array
+    def _unflatten_array(self, array, extra_dims):
+        return array.T.reshape(extra_dims + self.shape, order="F")
```

Comment:
`_flatten_array` and `_unflatten_array` are identical to the versions above, so the same comments apply. Moreover, perhaps this is reason enough to have a common baseclass? `size` can be made identical, I believe, with `prod(self.shape)` as well.